src/inventory.cpp

```diff
@@ -1070,6 +1070,20 @@ void inventory::update_invlet( item &newit, bool assign_invlet )
     }
 }
 
+void inventory::set_stack_favorite( const int position, const bool favorite )
+{
+    int p = 0;
+    for( auto &stack : items ) {
+        if( p == position ) {
```

Comment:
`std::advance()` could've done the trick in a much cleaner way.

Reply:
I'm not too familiar with the STL yet, thanks for the tip! I ended up using `std::next` for the same purpose.

https://github.com/CleverRaven/Cataclysm-DDA/blob/86a8b02d8219e76fac60da640a9d7514e2e51dd0/src/inventory.cpp#L1075-L1077